src/SprykerEco/Zed/Computop/Business/Hook/Mapper/Init/InitPayuCeeSingleMapper.php

```diff
@@ -0,0 +1,105 @@
+ */
+
+namespace SprykerEco\Zed\Computop\Business\Hook\Mapper\Init;
+
+use Generated\Shared\Transfer\QuoteTransfer;
+use Spryker\Shared\Kernel\Transfer\TransferInterface;
+use SprykerEco\Shared\Computop\ComputopConfig;
+use SprykerEco\Shared\Computop\Config\ComputopApiConfig;
+
+class InitPayuCeeSingleMapper extends AbstractMapper
+{
+    /**
+     * @return string
+     */
+    public function getMethodName(): string
+    {
+        return ComputopConfig::PAYMENT_METHOD_PAYU_CEE_SINGLE;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\QuoteTransfer $quoteTransfer
+     * @param \Spryker\Shared\Kernel\Transfer\TransferInterface $computopPaymentTransfer
+     *
+     * @return \Spryker\Shared\Kernel\Transfer\TransferInterface
+     */
+    public function updateComputopPaymentTransfer(
+        QuoteTransfer $quoteTransfer,
+        TransferInterface $computopPaymentTransfer
+    ): TransferInterface {
+        /** @var \Generated\Shared\Transfer\ComputopPayuCeeSinglePaymentTransfer $computopPaymentTransfer */
+        $computopPaymentTransfer = parent::updateComputopPaymentTransfer($quoteTransfer, $computopPaymentTransfer);
+        $computopPaymentTransfer
+            ->setMerchantId($this->config->getMerchantId())
+            ->setAmount($quoteTransfer->getTotals()->getGrandTotal());
+
+        $requestTransfer = $this->createRequestTransfer($computopPaymentTransfer);
+        $encryptedMac = $this->computopApiService->generateEncryptedMac($requestTransfer);
+        $computopPaymentTransfer->setMac($encryptedMac);
+
+        $decryptedValues = $this->computopApiService->getEncryptedArray(
+            $this->getDataSubArray($computopPaymentTransfer),
+            $this->config->getBlowfishPass()
+        );
+
+        if (!isset($decryptedValues[ComputopApiConfig::LENGTH], $decryptedValues[ComputopApiConfig::DATA])) {
+            return $computopPaymentTransfer;
+        }
+
+        $urlToComputop = $this->getUrlToComputop(
+            $computopPaymentTransfer->getMerchantId(),
+            $decryptedValues[ComputopApiConfig::DATA],
+            $decryptedValues[ComputopApiConfig::LENGTH]
+        );
+
+        $computopPaymentTransfer
+            ->setData($decryptedValues[ComputopApiConfig::DATA])
+            ->setLen($decryptedValues[ComputopApiConfig::LENGTH])
+            ->setUrl($urlToComputop);
+
+        return $computopPaymentTransfer;
+    }
+
+    /**
+     * @return string|null
+     */
+    protected function getActionUrl(): ?string
+    {
+        return $this->config->getPayuCeeSingleInitAction();
+    }
+
+    /**
+     * @param \Spryker\Shared\Kernel\Transfer\TransferInterface $computopPayuCeeSinglePaymentTransfer
+     *
+     * @return array
+     */
+    protected function getDataSubArray(TransferInterface $computopPayuCeeSinglePaymentTransfer): array
+    {
+        /** @var \Generated\Shared\Transfer\ComputopPayuCeeSinglePaymentTransfer $computopPayuCeeSinglePaymentTransfer */
```

Comment:
* You can move it to the method parameter declaration part.

Reply:
++